lib/handlers/authorize-handler.js

```diff
@@ -77,7 +77,12 @@ AuthorizeHandler.prototype.handle = function(request, response) {
     throw new InvalidArgumentError('Invalid argument: `response` must be an instance of Response');
   }
 
-  if ('false' === request.query.allowed) {
+  const notAllowed = [
+    request.query.allowed,
+    request.body.allowed
+  ].some(allowed => 'false' === allowed);
+
+  if (notAllowed) {
```

Comment:
Nah. It should be something like 

if (request.query.allowed === 'false' || request.body.allowed === 'false')

Also why is there no check for request.body.allowed === false? Does formData not allow boolean values?

Also: is it safe? Doesn't OAUth2 server add the allowed url fragment when the authorization finished. So actually you would get a redirect which is a Get and not a post. Not that we accidently make it possible to bypass something? Seems unlikely.

Reply:
Okay then I would update it towards this:

```js
const notAllowed = ['false', false].some(value => request.query.allowed === value || request.body.allowed === value)
```

Reply:
FormData does indeed not allow booleans. They only allow strings and blobs.

Reply:
Why do you like `some()`? Also query can not be a boolean value, as query has by definition only string values. 

```javascript
if (
    request.query.allowed === 'false' ||
    request.body.allowed === 'false' ||
    request.body.allowed === false
)
```

Is much easier to read and probably the most performant too. 

Just wanted to also suggest, that we should actually check for the opposite. Like query.allowed !== 'true'. From security point of view an explicit allowed is better.

Reply:
@jorenvandeweyer 
UrlEncoded FormData, Thanks for looking up. 

So it should be simply:

```javascript
if (
    request.query.allowed === 'false' ||
    request.body.allowed === 'false'
)
```

Reply:
Ok, will do